.travis.yml

```diff
@@ -1,76 +1,100 @@
+        wget 
+        curl && 
+    source /opt/ros/kinetic/setup.bash && 
+    catkin_init_workspace && 
+    cd ~/catkin_ws && 
+    catkin build && 
+    source devel/setup.bash && 
+    cd ~/catkin_ws/src && 
+    ln -s /travis ."
 
 # Install all dependencies, using wstool first and rosdep second.
 # wstool looks for a ROSINSTALL_FILE defined in the environment variables.
 before_script:
-  # source dependencies: install using wstool.
-  - cd ~/catkin_ws/src
-  - wstool init
-  - if [[ -f $ROSINSTALL_FILE ]] ; then wstool merge $ROSINSTALL_FILE ; fi
-  - wstool up
-  # package depdencies: install using rosdep.
-  - cd ~/catkin_ws
-  - rosdep install -y --from-paths src --ignore-src --rosdistro $ROS_DISTRO
+  - docker exec -t alica-test bash -c "
+    cd ~/catkin_ws/src && 
+    wstool init && 
+    if [[ -f /travis/dependencies.rosinstall ]] ; then 
+        sed -i -e \"s/rr-devel/$BRANCH/g\" /travis/dependencies.rosinstall; 
+        if wstool merge -y /travis/dependencies.rosinstall ; then 
+            sed -i -e \"s/$BRANCH/rr-devel/g\" /travis/dependencies.rosinstall; 
+            wstool merge -y /travis/dependencies.rosinstall; 
+        fi; 
+    fi && 
+    wstool up && 
+    cd ~/catkin_ws && 
+    rosdep install -y --from-paths src --ignore-src --rosdistro kinetic"
 
 # Compile and test (mark the build as failed if any step fails). If the
 # CATKIN_OPTIONS file exists, use it as an argument to catkin_make, for example
 # to blacklist certain packages.
 #
+# If the current branch is coverity_scan, trigger a coverity analysis.
+#
 # NOTE on testing: `catkin_make run_tests` will show the output of the tests
 # (gtest, nosetest, etc..) but always returns 0 (success) even if a test
 # fails. Running `catkin_test_results` aggregates all the results and returns
 # non-zero when a test fails (which notifies Travis the build failed).
 script:
-  - source /opt/ros/$ROS_DISTRO/setup.bash
-  - cd ~/catkin_ws
-  - catkin build --no-status $( [ -f $CATKIN_OPTIONS ] && cat $CATKIN_OPTIONS )
-  # Run the tests, ensuring the path is set correctly.
-  - source devel/setup.bash
-  - catkin run_tests && catkin_test_results
+  - docker exec -t alica-test bash -c "
+    source /opt/ros/kinetic/setup.bash && 
+    cd ~/catkin_ws && 
+    catkin build --no-status && 
+    source devel/setup.bash && 
+    catkin run_tests --no-status && catkin_test_results && 
+    if [[ \"$BRANCH\" = \"coverity_scan\" ]] ; then 
+        wget -qO- https://scan.coverity.com/download/cxx/linux64 
+            --post-data \"token=o6qYj0vj-t-K1PKQj5ugyg&project=rapyuta-robotics/alica\" | 
+        tar xvz && 
+        python /travis/generate_coverity-analysis_script.py . && 
+        chmod +x coverity-analysis.sh && 
+        ./cov-analysis*/bin/cov-build --dir cov-int ./coverity-analysis.sh && 
+        tar czvf cs.tgz cov-int && 
+        curl 
+            --form token=o6qYj0vj-t-K1PKQj5ugyg 
+            --form email=alexander.bolinsky@rapyuta-robotics.com 
```

Comment:
general rapyuta account, it might make sense to make this an environment variable.

Reply:
If this is going to be merged upstream it makes sense to comeup with a general email account used by the Alica team.

Reply:
Made them environment variables!